### PR TITLE
[CIT-561] Add aarch64 and arm64 support to install script

### PR DESCRIPTION
**Pull request by makr11st**
## Proposed changes

Add support for `aarch64` and `arm64` architectures in the install script.
Adding the download files to the release is required before merging.

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by albinsuresh — APPROVED** [review]

File: get-thin-edge_io.sh [hunk 1]
```diff
@@ -38,9 +38,21 @@ fi
 echo "${BLUE}Installing mosquitto as prerequirement for thin-edge.io${COLORRESET}"
 apt install mosquitto -y
 
+# Packages for `arm64` are zipped in files with `aarch64` in names.
```
[the remark this answers is not in the document]
Comment (makr11st) [reply get-thin-edge_io.sh:41]: I see the comment doesn't give the actual situation the justice.
Your observation is correct, there are no aarch64/arm64 packages in the release at the moment, that's why I made a comment in the description that it shouldn't be merged before those files are merged, additionally there is no aarch64/arm64 packages available before 0.3.0, therefore I think an additional check for version should be applied, but I need to discuss this with Sebastian before, as this is not in the ticket.

[the remark this answers is not in the document]
Comment (albinsuresh) [reply get-thin-edge_io.sh:41]: Okay. I thought the same `amd64` binaries can be used even on the systems reporting `aarch64` and `arm64`. So, we're going to create new packages for these archs as well, named after them?

[the remark this answers is not in the document]
Comment (itsyitsy) [reply get-thin-edge_io.sh:41]: Having aarch64/arm64 support only after version 0.3.0 seems ok. Just uploaded the arm64 package to the latest release

File: get-thin-edge_io.sh [hunk 2]
```diff
@@ -38,9 +38,21 @@ fi
...
+if [ "$ARCH" = "arm64" ]
+then
+    ARCH='aarch64'
+fi
+
```
Comment (itsyitsy) [inline get-thin-edge_io.sh:46]: As discussed in our chat this morning

File: get-thin-edge_io.sh [hunk 3]
```diff
@@ -38,9 +38,21 @@ fi
...
 wget https://github.com/thin-edge/thin-edge.io/releases/download/${VERSION}/tedge_${VERSION}_${ARCH}.deb -P /tmp/tedge
 wget https://github.com/thin-edge/thin-edge.io/releases/download/${VERSION}/tedge_mapper_${VERSION}_${ARCH}.deb -P /tmp/tedge
 
+# Packages for `aarch64` contain *.deb files with `arm64` in names.
+if [ "$ARCH" = "aarch64" ]
+then
+    ARCH='arm64'
+fi
+
```
Comment (itsyitsy) [inline get-thin-edge_io.sh:55]: Please remove this, also as discussed

**Review by itsyitsy — CHANGES_REQUESTED** [review]

File: get-thin-edge_io.sh [hunk 4]
```diff
@@ -17,16 +17,31 @@ echo "${BLUE}Thank you for trying thin-edge.io! ${COLORRESET}\n"
...
-if [ "$ARCH" = "armhf" ] || [ "$ARCH" = "amd64" ]
+if [ "$ARCH" = "aarch64" ] || [ "$ARCH" = "arm64" ] || [ "$ARCH" = "armhf" ]  || [ "$ARCH" = "amd64" ]
 then
+    # Some OSes may read architecture type as `aarch64`, `aarch64` and `arm64` are the same architectures types.
+    if [ "$ARCH" = "aarch64" ]
+    then
+        ARCH='arm64'
+    fi
+
+    # For arm64 only version versions above 0.3.0 are available.
```
Comment (albinsuresh) [inline get-thin-edge_io.sh:35]:
```suggestion
    # For arm64, only the versions above 0.3.0 are available.
```

File: get-thin-edge_io.sh [hunk 5]
```diff
@@ -17,16 +17,31 @@ echo "${BLUE}Thank you for trying thin-edge.io! ${COLORRESET}\n"
...
+    if [ "$ARCH" = "arm64" ] && ! dpkg --compare-versions "$VERSION" ge "0.3.0"
+    then
+        echo "aarch64/arm64 can only be installed with version 0.3.0 or above."
```
Comment (albinsuresh) [inline get-thin-edge_io.sh:38]:
```suggestion
        echo "aarch64/arm64 compatible packages are only available for version 0.3.0 or above."
```

**Review by albinsuresh — APPROVED** [review]

**Review by itsyitsy — APPROVED** [review]

**Pull request merged**